Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabSaveLoadHandler.cpp

```diff
@@ -993,5 +1020,59 @@ namespace AzToolsFramework
 
             return AZStd::move(unsavedPrefabsContainer);
         }
+
+        void PrefabSaveHandler::SourceFileRemoved(
+            AZStd::string relativePath, AZStd::string scanFolder, [[maybe_unused]] AZ::Uuid sourceUUID)
+        {
+            // This gets triggered for every source file. We only need source files that are prefabs and are loaded in the current level.
+            TemplateId loadedTemplateId = s_prefabSystemComponentInterface->GetTemplateIdFromFilePath(relativePath.c_str());
+            if (loadedTemplateId != InvalidTemplateId)
+            {
+                InstanceSetConstReference instancesMappedToTemplate =
+                    m_templateInstanceMapperInterface->FindInstancesOwnedByTemplate(loadedTemplateId);
+
+                if (!instancesMappedToTemplate.has_value() || instancesMappedToTemplate->get().size() == 0)
+                {
+                    // There are no instances mapping to this template. So it can be removed from memory.
+                    s_prefabSystemComponentInterface->RemoveTemplate(loadedTemplateId);
+                }
+                else
+                {
+                    TemplateId rootTemplateId = m_prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
+                    if (rootTemplateId == loadedTemplateId)
+                    {
+                        // Deletion of the current level that the user has opened needs to be handled differently. Skip for now
+                        return;
+                    }
+                    else
+                    {
+                        AzFramework::EntityContextId editorEntityContextId = AzFramework::EntityContextId::CreateNull();
+                        AzToolsFramework::EditorEntityContextRequestBus::BroadcastResult(
+                            editorEntityContextId, &AzToolsFramework::EditorEntityContextRequestBus::Events::GetEditorEntityContextId);
+
+                        if (m_prefabFocusPublicInterface->GetFocusedPrefabTemplateId(editorEntityContextId) == loadedTemplateId)
+                        {
+                            m_prefabFocusPublicInterface->FocusOnParentOfFocusedPrefab(editorEntityContextId);
+                        }
+
+                        auto button = QMessageBox::question(
+                            AzToolsFramework::GetActiveWindow(),
+                            QString(),
+                            QString("Prefab '%1' has been deleted from the file system. But we detected %2 instance(s) of it "
+                                    "in the level. Would you like to delete them ?")
```

Comment:
If there are no instances instantiated in Outliner, the dialog would still show up with "0 instance(s)"?

Reply:
No. When there are 0 instances, we will remove the template from memory without notifying the user. It's handled here : https://github.com/o3de/o3de/pull/14197/files#diff-d5729d094928c4642299dddca98bba2de13aec0c7220ead69fb2f42f949fb751R1034

Reply:
Cool. I missed that line. Thanks!